### PR TITLE
384 | Refactors validation and introduces test for validate_config()

### LINKED ISSUE #384: Refactor Validation
Currently validation has grown from #302 such that I've written individual functions for each schema that is being validated. This is poor design. Instead there should be a single validation function and `Schemas` should be passed in for validation.

### PR DESCRIPTION
**Pull request by ns-rse**
Closes #384

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (SylviaWhittle) [inline .pre-commit-config.yaml:33]: Does this mean that `.ipynb` notebook files will be checked with pylint ?

File: .pylintrc
Comment (SylviaWhittle) [inline .pylintrc:176]: Just wondering, why remove the FIXME from the notes that pylint warns about?

File: topostats/run_topostats.py
Comment (SylviaWhittle) [inline topostats/run_topostats.py:293]: I assume there is a reason for doing so, but why iterate over `items()` instead of `keys()` if only the key is to be used?

**Review by SylviaWhittle — APPROVED** [review]
Looks good. A couple of minor comments / questions. 

I ensured that the config validation works and that the `test_validation.py` picks up errors that break functionality.

Thank you!

File: .pylintrc
Comment (ns-rse) [reply .pylintrc:176]: I've removed this as there are a few `FIXME` notes around in the codebase that do need addressing. They should really be written up into issues, but would likely get overlooked as specific tasks as other work takes priority but by leaving them in place they act as an _aide-mémoire_ when myself or anyone else is looking at/working on that section of code.

File: topostats/run_topostats.py
Comment (ns-rse) [reply topostats/run_topostats.py:293]: Yep, its [pylint consider-using-dict-items C0206](https://pylint.pycqa.org/en/latest/user_guide/messages/convention/consider-using-dict-items.html).

File: .pre-commit-config.yaml
Comment (ns-rse) [reply .pre-commit-config.yaml:33]: Ah, good spot. I didn't copy over the necessary change to `setup.cfg` from #398 which excludes `ipynb` from `pylint`, once that is merged it will sort things it.

It was because #398 was awaiting approval and I wanted to get this quick improvement to validation out for PR that I copied the changes over, need to be more attentive next time.

**Pull request merged**
